Automattic/wp-calypso
# Hosting Command Palette: Add tracks to command selection and site selection

- Related to https://github.com/Automattic/dotcom-forge/issues/4670

## Proposed Changes

* It adds 5 events. Open, Select a command, Select a site, Close and Go back to root commands.
* I'll create the events on the tracks repository after I get approved this PR.

Here are a real example of the properties for each event:

#### `calypso_hosting_command_palette_open`:
  * `current_route: "/home/:site"` 
#### `calypso_hosting_command_palette_command_select`:
  * `current_route : "/home/:site"`
  * `command : "openSiteDashboard"`
  * `has_nested_commands : true`
  * `list_count : 56`
  * `list_visible_count : 2`
  * `search_text : "dashbo"`
#### `calypso_hosting_command_palette_site_select`:
  * `current_route : "/home/:site"`
  * `command : "openSiteDashboard"`
  * `has_nested_commands : false`
  * `list_count : 140`
  * `list_visible_count : 140`
  * `search_text : ""`
  * `command_site_id : 225687375`
  * `command_site_has_custom_domain : false`
  * `command_site_plan_id : 1018`
#### `calypso_hosting_command_palette_back_to_root` :
  * `current_route : "/home/:site"`
  * `command : "enableEdgeCache"`
  * `from_keyboard : true`
  * `search_text : ""`
#### `calypso_hosting_command_palette_close`:
  * `current_route : "/home/:site"`
  * `command : ""`
  * `search_text : ""`

Tracks will add other properties on top of these, such as:
- `blog_id`, `blog_lang`, `client`, `environment`, `environment_id`, `site_count`, `site_id_label`, `site_plan_id`, `vph`, `vpw`.

## Testing Instructions



* Enable event logs in Chrome dev tools: `localStorage.setItem( 'debug', 'calypso:analytics*' );` or use the vigilante chrome extension

* Open the command palette by pressing `cmd+k`.
  * Observe the event `calypso_hosting_command_palette_open` is fired.
* Select a root command with no nested sites, like `Open help`
  * Observe the event `calypso_hosting_command_palette_command_select` is fired with `has_nested_commands=false`
* Select a root command with nested sites, like `Open site dashbaord...`
  * Observe the event `calypso_hosting_command_palette_command_select` is fired with `has_nested_commands=true`
* Select a site
  * Observe the event `calypso_hosting_command_palette_site_select` is fired. 
* Open the command palette by pressing `cmd+k`.
* Select a root command with nested sites, like `Open site dashbaord...`
* Go back
  * Observe the event `calypso_hosting_command_palette_back_to_root` is fired. 
* Close the Command palette by pressing `esc`
  * Observe the event `calypso_hosting_command_palette_close` is fired. 

* Observe all the respective event were fired


## Screencast



https://github.com/Automattic/wp-calypso/assets/779993/5a6368db-2882-41d2-9ce1-f175f709f7de



## Pre-merge Checklist
